source/Playlist.cpp

```diff
@@ -0,0 +1,133 @@
+/* Playlist.cpp
+ Copyright (c) 2022 by RisingLeaf
+
+ Endless Sky is free software: you can redistribute it and/or modify it under the
+ terms of the GNU General Public License as published by the Free Software
+ Foundation, either version 3 of the License, or (at your option) any later version.
+
+ Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+ WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License along with
+ this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
```

Comment:
Needs `#include <algorithm>` or it'll fail to compile with g++ due to the algorithm find being absent. (It tries to use a string find function instead.)